### PR TITLE
Mongodb backend

### PR DESCRIPTION
**Pull request by bbotella**

### CODE REVIEW

**Comment by sibiryakov** [comment]
Hey @bbotella you did great job implementing Mongodb backend for CF!

**Comment by sibiryakov** [comment]
Could you please remove the files which aren't related to Mongodb backend and rebase the branch to latest development?

File: crawlfrontier/contrib/backends/mongo.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+        # Log
+        self.manager.logger.backend.debug('PAGE_CRAWLED page=%s status=%s links=%s' %
+                                          (response, response.status_code, len(links)))
+
+        # process page crawled
+        backend_page = self._page_crawled(response)
+
+        # Update crawled fields
+        backend_page.state = self.State.CRAWLED
+        self.collection.update({'_meta.fingerprint': backend_page._meta['fingerprint']}, {
```
Comment (sibiryakov) [inline crawlfrontier/contrib/backends/mongo.py]: May be it makes sense also to make a key customizable by subclasses?

File: crawlfrontier/contrib/backends/mongo.py [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+        ERROR = 'ERROR'
+
+    def __init__(self, manager):
+        settings = manager.settings
+        mongo_ip = settings.get('BACKEND_MONGO_IP', None)
+        mongo_port = settings.get('BACKEND_MONGO_PORT', None)
+        if mongo_ip is None or mongo_port is None:
+            raise NotConfigured
+        self.client = MongoClient(mongo_ip, mongo_port)
+        self.database_name = ''.join(
```
Comment (sibiryakov) [inline crawlfrontier/contrib/backends/mongo.py]: This kind of logic is pretty specific. User could do the same in settings.py file, if such behavior is really needed.

File: crawlfrontier/contrib/backends/mongo.py [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+        if created:
+            backend_page.created_at = now
+
+        # Update fields
+        backend_page.last_update = now
+        backend_page.status = response.status_code
+        return backend_page
+
+    def _request_to_mongo(self, request):
+        return json.loads(json.dumps(vars(request), sort_keys=True, indent=4, default=json_util.default),
```
Comment (sibiryakov) [inline crawlfrontier/contrib/backends/mongo.py]: Is there are the way how to make it computationally cheaper?

**Comment by sibiryakov** [comment]
Closing in favor of https://github.com/scrapinghub/crawl-frontier/pull/17

**Comment by dingld** [comment]
It seems that mongodb backend is no longer supported in frontera?

**Comment by sibiryakov** [comment]
@dingld it requires some work, you could take it from #17 and try to make it work.

**Pull request closed without merging**